tests/different-conditions-strict/Makefile

```diff
@@ -0,0 +1,20 @@
+include $(dir $(MAKEFILE_LIST))../common.mk
+
+all:
+	mkdir -p build1 build2
+	cd build1 ; cat ../main_hdr.c ../main1.c ../main_ftr.c > ../main.c ; $(CC) $(CXXFLAGS) ../main.c -o testcase
+	cd build2 ; cat ../main_hdr.c ../main2.c ../main_ftr.c > ../main.c ; $(CC) $(CXXFLAGS) ../main.c -o testcase
+
+run: json
+
+json:
+	cd build1 ; ./testcase
+	cd build2 ; ./testcase
+	$(GCOVR) --verbose --merge-mode-conditions=strict --json-pretty --json > coverage.log 2>&1 ; test $$? -eq 64 | cat coverage.log
+	cat coverage.log
+	touch empty
```

Comment:
For what is this needed?

Reply:
I could not figure out how to run a test without any reference data intially.

I assume it runs the test against all compilers then and only checks against reference data if some is defined for the test?

Reply:
Yes. But it searches the reference data also in older versions. If e.g. `coverage.json` isn't found for `gcc-14-Darwin` it is searched in `gcc-14`, `gcc-13-Darwin`, `gcc-13`, ...